### PR TITLE
[PKG-3770] 4.1.0

### PR DESCRIPTION
**Pull request by ViridianMelody**
pytorch-tabnet 4.1.0 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-3770](https://anaconda.atlassian.net/browse/PKG-3770) 
- [Upstream repository](https://github.com/dreamquark-ai/tabnet)
- [Upstream changelog/diff](https://github.com/dreamquark-ai/tabnet/releases/tag/v4.1.0)

### Explanation of changes:

- Added upstream tests.
- Matched dependency pinnings to upstream.

[PKG-3770]: https://anaconda.atlassian.net/browse/PKG-3770?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -7,46 +7,52 @@ package:
...
+  skip: True  # [py<37]
+  script: {{ PYTHON }} -m pip install . -vv --no-build-isolation --no-deps
 
 requirements:
   host:
     - pip
-    - python >=3.6
+    - wheel
+    - python
     - poetry >=0.12
```
Comment (cbouss) [inline recipe/meta.yaml:23]:
```suggestion
    - poetry-core >=0.12
```
That's all that is actually needed to build.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -7,46 +7,52 @@ package:
...
   run:
-    - python >=3.6
-    - numpy >=1.17,<2.0
+    - python
+    - numpy {{ numpy }}
```
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
    - numpy >=1.17
```
Using the cbc version is for when numpy is used in host and the C-API is used. Here we can stick to what upstream defines.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -7,46 +7,52 @@ package:
...
-  noarch: python
-  script: {{ PYTHON }} -m pip install . -vv
+  skip: True  # [py<37]
+  script: {{ PYTHON }} -m pip install . -vv --no-build-isolation --no-deps
 
 requirements:
   host:
     - pip
-    - python >=3.6
+    - wheel
```
Comment (cbouss) [inline recipe/meta.yaml:21]:
```suggestion
```
If using poetry-core, I think the linter will not flag it.

File: recipe/meta.yaml (hunk 1 above)
Comment (ViridianMelody) [reply recipe/meta.yaml:23]: I actually tried that, but it failed: `ModuleNotFoundError: No module named 'poetry.masonry'`

Comment (cbouss) [reply recipe/meta.yaml:23]: Interesting, apparently changing to poetry-core, the backend should be updated to poetry.core.masonry.api in pyproject.toml.
But nevermind as it works with poetry.

**Review by cbouss — APPROVED** [review]

**Review by SC426 — APPROVED** [review]

**Pull request merged**